src/main/java/com/google/sps/data/TextAnalyser.java

```diff
@@ -210,45 +208,86 @@ public AnalyzeSyntaxResponse analyseSyntaxText() throws IOException {
     }
   }
 
-  public Set<String> getAdjectives() throws IOException {
+  // Get the combination of verb-adverb
+  public Set<String> getSyntax() throws IOException {
     Set<String> adjectives = new LinkedHashSet<String>();
+    String verb = "";
+    boolean add = false;
 
     for (Token token : analyseSyntaxText().getTokensList()) {
       String partOfSpeech = token.getPartOfSpeech().getTag().toString();
-      if(partOfSpeech.equals("ADJ")) {
-        adjectives.add(token.getLemma().toLowerCase());
+      
+      if(partOfSpeech.equals("VERB") && !add) {
+        verb = token.getText().getContent().toLowerCase();
+        add = true;
+      }
+
+      if(partOfSpeech.equals("ADV") && add) {
+        adjectives.add(verb + " " + token.getLemma().toLowerCase());
+        add = false;
       }
     }
 
     return adjectives;
   }
 
-  public String checkInjection() {
+  public boolean isInjection() {
     if (message.indexOf("<script>") != -1 || message.indexOf("</script>") != -1 ||
         message.indexOf("<html>") != -1 || message.indexOf("</html>") != -1) {
-      return "html-injection";
+      return true;
     }
 
-    return "no-html-injection";
+    return false;
   }
 
-  // put all the key words together
-  // use a LinkedHashSet to remove duplicates but maintain order
+  // Put all the key words together
+  // Use a LinkedHashSet to remove duplicates but maintain order
   public Set<String> getKeyWords() {
     try {
       Set<String> keyWords = new LinkedHashSet<String>();
+
       keyWords.addAll(getGreetings());
-      keyWords.addAll(getEntities());
       keyWords.addAll(getEvents());
-      keyWords.addAll(getAdjectives());
+      keyWords.addAll(getEntities());
       keyWords.addAll(getCategories());
       keyWords.add(getMood());
 
       return keyWords;
     } catch (IOException e) {
-      // no key words  
+      // No key words  
       System.err.println("There are no key words.");
+      e.printStackTrace();
       return Collections.emptySet();
     }
   }
+
+  public Set<String[]> getSetsOfKeyWords() {
+    Set< String[] > setsOfKeyWords = new LinkedHashSet<>();
+    List<String> keyWords = convertToList(getKeyWords());
+
+    // Only returns one key word when only a sentiment is found
+    if(keyWords.size() == 1) {
+      setsOfKeyWords.add(new String[] {keyWords.get(0)});
+      return setsOfKeyWords;
+    }
+
+    /**
+    * TODO: Add default value when no key word is found
+    * and discuss with team
+    **/
+
+    for (int i = 0; i < keyWords.size(); i++) {
+      for (int j = i + 1; j < keyWords.size(); j++) {
+        setsOfKeyWords.add(new String[] {keyWords.get(i), keyWords.get(j)});
+      }
+    }
+
+    return setsOfKeyWords;
+  }
+
+  // Generic function to convert set to list
+  public static List<String> convertToList(Set<String> set)
```

Comment:
Maybe that method isn't needed, it's just one line and `new ArrayList(set)` pretty much says it all.